tests/python/unittest/test_tir_ir_builder.py

```diff
@@ -554,6 +554,147 @@ def check_target(target):
         check_target(target)
 
 
+@tvm.testing.requires_gpu
```

Comment:
I don't think that the dyn_shared_mem tests should be in the `test_tir_ir_builder.py`. It would be great if we can create a separate file to test the pass `tir.transform.MergeDynamicSharedMemoryAllocations` by comparing the transformed IR and the expected IR.

Reply:
Also, is it possible to write the test using TVMScript? I think it's more clear than the `ir_builder`

Reply:
Thanks for a good suggestion. Moved new test cases to `test_tir_transform_merge_dynamic_shared_memory_allocations.py` and added verification on the output of the new pass.

I'm not familiar with TVMScript (also never liked hybrid script). Now that I've written the test using IR builder, I don't have a big motivation to do rewrite.  But I'll look over existing TVMScript examples and see if I can do the rewrite quickly.